src/Microsoft.ML.ImageAnalytics/MLImage.cs

```diff
@@ -0,0 +1,441 @@
+            }
+
+            return new MLImage(image);
+        }
+
+        /// <summary>
+        /// Creates MLImage object from the pixel data span.
+        /// </summary>
+        /// <param name="width">The width of the image in pixels.</param>
+        /// <param name="height">The height of the image in pixels.</param>
+        /// <param name="pixelFormat">The pixel format to create the image with.</param>
+        /// <param name="imagePixelData">The pixels data to create the image from.</param>
+        /// <returns>MLImage object.</returns>
+        public static unsafe MLImage CreateFromPixels(int width, int height, MLPixelFormat pixelFormat, ReadOnlySpan<byte> imagePixelData)
+        {
+            if (pixelFormat != MLPixelFormat.Bgra32 && pixelFormat != MLPixelFormat.Rgba32)
+            {
+                throw new ArgumentException($"Unsupported pixel format", nameof(pixelFormat));
+            }
+
+            if (imagePixelData.Length != width * height * 4)
+            {
+                throw new ArgumentException($"Invalid {nameof(imagePixelData)} buffer size.");
+            }
+
+            SKBitmap image = new SKBitmap(new SKImageInfo(width, height, pixelFormat == MLPixelFormat.Bgra32 ? SKColorType.Bgra8888 : SKColorType.Rgba8888));
+
+            Debug.Assert(image.Info.BitsPerPixel == 32);
+            Debug.Assert(image.RowBytes * image.Height == width * height * 4);
+
+            imagePixelData.CopyTo(new Span<byte>(image.GetPixels().ToPointer(), image.Width * image.Height * 4));
+
+            return new MLImage(image);
+        }
+
+        /// <summary>
+        /// Gets the pixel format for this Image.
+        /// </summary>
+        public MLPixelFormat PixelFormat { get; private set; }
+
+        /// <summary>
+        /// Gets the image pixel data.
+        /// </summary>
+        public ReadOnlySpan<byte> Pixels
+        {
+            get
+            {
+                ThrowInvalidOperationExceptionIfDisposed();
+                Debug.Assert(_image.Info.BytesPerPixel == 4);
+
+                return _image.GetPixelSpan();
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the image tag.
+        /// </summary>
+        public string Tag { get; set; }
+
+        /// <summary>
+        /// Gets the image width in pixels.
+        /// </summary>
+        public int Width
+        {
+            get
+            {
+                ThrowInvalidOperationExceptionIfDisposed();
+                return _image.Width;
+            }
+        }
+
+        /// <summary>
+        /// Gets the image height in pixels.
+        /// </summary>
+        public int Height
+        {
+            get
+            {
+                ThrowInvalidOperationExceptionIfDisposed();
+                return _image.Height;
+            }
+        }
+
+        /// <summary>
+        /// Gets how many bits per pixel used by current image object.
+        /// </summary>
+        public int BitsPerPixel
```

Comment:
Is this now redundant to PixelFormat?

Reply:
It is super helpful to consume when allocating buffers for the pixel colors. It saves checking the PixelFormat. I am just thinking in the future we can extend our interfaces which can accommodate pixel formats that have color not using byte boundaries (i.e. using bits instead). I'll keep this property as I am seeing it helpful but let mem know if you disagree.